azure-iot-device/azure/iot/device/iothub/pipeline/upload_pipeline.py

```diff
@@ -0,0 +1,91 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import sys
+from azure.iot.device.common.evented_callback import EventedCallback
+from azure.iot.device.common.pipeline import (
+    pipeline_stages_base,
+    pipeline_ops_base,
+    pipeline_stages_http,
+)
+from . import (
+    constant,
+    pipeline_stages_iothub,
+    pipeline_ops_iothub,
+    pipeline_ops_upload,
+    pipeline_stages_upload_http,
+)
+from azure.iot.device.iothub.auth.x509_authentication_provider import X509AuthenticationProvider
+
+logger = logging.getLogger(__name__)
+
+
+class UploadPipeline(object):
+    """Pipeline to communicate with Upload endpoint of IoT Hub.
+    Uses HTTP.
+    """
+
+    def __init__(self, auth_provider, pipeline_configuration):
+
+        # Event Handlers - Will be set by Client after instantiation of this object
+        self.on_connected = None
+        self.on_disconnected = None
+        self.on_c2d_message_received = None
+        self.on_input_message_received = None
+        self.on_method_request_received = None
+        self.on_twin_patch_received = None
+
+        self._pipeline = (
+            pipeline_stages_base.PipelineRootStage(pipeline_configuration=pipeline_configuration)
```

Comment:
As with the method invoke pipeline, aren't there some other stages we want here?

Reply:
no

---
In reply to: [351491895](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r351491895) [](ancestors = 351491895)